### ISSUE TITLE
procedure to Install EditEmailSubject-MX dev version on Thunderbird 68+ ?

### ISSUE THREAD

**Issue by Sylvain303** [cleidigh/EditEmailSubject-MX#13]
Hi,

I would like to help and test this Great extension. How do I install the development version on my Thunderbird?

I can write down tutorial, example, screenshot, etc. in return.
Also doing bug reporting.

May be just show me how to build the xpi, and I will produce a documentation bootstrap.

Regards,
Sylvain.

**Comment by neandr** [comment]
Hi Sylvain,
any support for project is very much welcomed! Thanks for your interest.

The main focus of the project is to describe for addon authors what are the steps to move from a legacy TB60 version to Tb68/7X. As you see this project is very much in a start phase only.

The addon itself (with it current TB60 status) can be downloaded for here. If download a ZIP of the project you will find the required XPI (see the zip directory /EditEmailSubject-MX-master/editemailsubject-original/xpi/)
Also there is a branch 'dev_gN' which holds a few first steps on our way.
Alternatively the addon can also [be found on TB network](https://addons.thunderbird.net/en-US/thunderbird/addon/edit-email-subject/?src=ss)

**Comment by Sylvain303** [comment]
Hi Guenter,

Sorry, I'm too new to TB extension coding to find a working xpi file which I could install on TB 68.2.2 (64-bit) I'm running.

My description may be insufficient: I would like to install  EditEmailSubject on TB68.
Which doesn't work any more. 

* the addon editemailsubject-2.1.1-tb.xpi [on TB network](https://addons.thunderbird.net/en-US/thunderbird/addon/edit-email-subject/?src=ss) is failing
* also editemailsubject-2.1.1-tb.xpi [on dev_gN branch](https://github.com/cleidigh/EditEmailSubject-MX/blob/dev_gN/xpi/editemailsubject-2.1.2b1.xpi) is failing too

![Screenshot_2020-01-16_06-52-12_TB68_install_fail_editemailsubject-2 1 2b1 xpi](https://user-images.githubusercontent.com/274026/72497478-e7fc2780-382c-11ea-9c5e-fee40affaf4b.png)

I found [cmds.txt](https://github.com/cleidigh/EditEmailSubject-MX/blob/dev_gN/cmds.txt) which seems give some packaging instruction. Is that enough? I mean, just ziping stuff? No compilation, signing?

I suppose most of the code is in:

```
wc -l ./src/chrome/content/editemailsubject.js
335 ./src/chrome/content/editemailsubject.js
```

Which is quite short. 

I understand your goal to use this extension as an example to migrate from legacy extension to the new MailExtension format. So giving my newbee status about TB extension. Go for it.

Steps draft: (could be edited in the Wiki or a separate MarkDown file)

1. Read this doc (new format description) ([may be here](https://developer.thunderbird.net/add-ons/tb68) or [this one](https://developer.thunderbird.net/add-ons/tb68/bootstrapped) or [HelloWord](https://developer.thunderbird.net/add-on-examples/hello-world-add-on))
2. In your repository create a new branch
3. copy stuff from here to here
4. change this part following those rules (may be an helper to produce the new structure) (ex: manifest.json)
5. build the extension with this script (some bash helper, or windows .bat or such)
6. try install the xpi from the Add-ons Manager (graphical steps, some are provided in debug mode in  [HelloWord](https://developer.thunderbird.net/add-on-examples/hello-world-add-on))

etc.

I tried to follow Debug steps found in  [HelloWord](https://developer.thunderbird.net/add-on-examples/hello-world-add-on) loading `./EditEmailSubject-MX/EditMailSubject-MX/webextension/manifest.json` but nothing seems to happen.

Hope that more clear. (for me a bit more)
Regards,
Sylvain.

**Comment by neandr** [comment]
Hi Sylvian,
please note what I said before:
> The addon itself (with it current TB60 status) can be downloaded for here
The current code you find is NOT working on TB68! It's for TB60!

We're in WIP here. And our whole project is to show to other addon developers which steps are necessary to get it ready for TB68/7++. For sure we will just show one way and there may be others. But all of them need a lot of technical conversions and they are more complex than going for cmds.txt or the HelloWorld example.

Stay tuned ;)

**Comment by Sylvain303** [comment]
Thanks,
I realized that the code is not working. It wasn't clear from my point of view.

I will document myself a bit more and try to figure out how to help, here. Especially this extension is so useful that it need to be fixed soon.

If you have any more link that can put me on the right path, post it here. I will document the issue subject, too. 

Regards,
Sylvain. (French Firstname, note the order of the vowels) :wink:

**Comment by jobisoft** [comment]
I started to make the addon compatible with TB68, which is the first step to get it compatible with 74 or later.

https://github.com/cleidigh/EditEmailSubject-MX/issues/17#issuecomment-597535305

As written in that issue, I have not fixed the IMAP storage mode, only the local storge mode and I still have a few questions which I raised in that issue. Maybe you can help out. 

Do you know how to build an XPI out of the [TB68 branch](https://github.com/cleidigh/EditEmailSubject-MX/tree/TB68)?

**Comment by Sylvain303** [comment]
Hi @jobisoft 

Thanks for the update. Unfortunately I'm currently too busy to find the time to extend my knowledge to understand a TB Extension.

No, I'm not aware of how to build a TB extension.
I'm currently using TB68.5.0 (64-bit) under xubuntu 18.04

Could you tell me?

I've post something above in https://github.com/cleidigh/EditEmailSubject-MX/issues/13#issuecomment-575007048 but I need to restart again, now.

**Comment by jobisoft** [comment]
You can download the current test version here:
https://github.com/cleidigh/EditEmailSubject-MX/raw/TB68/xpi/editemailsubject-3.0.0b2.xpi

Can you take a look at this issue, where I ask for confirmation, that this version is fully working?

**Comment by jobisoft** [comment]
Forgot to post the issue I was referring to:
https://github.com/cleidigh/EditEmailSubject-MX/issues/17#issuecomment-598051425

**Comment by jobisoft** [comment]
The release page now includes a link to an XPI file for each release to be downloaded. I also updated the main Readme which is loaded on the main repository page:

https://github.com/cleidigh/EditEmailSubject-MX

**Comment by neandr** [comment]
Would prefer not to use the
https://github.com/cleidigh/EditEmailSubject-MX/tree/master/xpi for
storing the XPI releases.
Githup has a "release" page:
https://github.com/cleidigh/EditEmailSubject-MX/releases
and the XPI history is stored there already.
/(Suggestion: delete the dir ..///EditEmailSubject-MX/tree/master/xpi
totally)/

Also the README.md
<https://github.com/cleidigh/EditEmailSubject-MX/blob/master/README.md>
instruction for installing the XPI should include the note about D&D of
the XPI from the /..///EditEmailSubject-MX/releases/ to the
TB/AddonManager page /(seems to be the easier way)//
/


Am 19.03.20 um 12:18 schrieb John Bieling:

**Comment by jobisoft** [comment]
The idea behind having them in the repository directly: If you clone it, you have the XPI as well. I added downloadlinks to the release page. So far, you where the only one, who added his XPI there directly :-)

I also added download instructions to the README, is that enough? What is missing?

**Comment by jobisoft** [comment]
@neandr : Furthermore, l prefer the download link in the description over the XPi in the assets, because users do not see the content of the assets until they click on them. My idea here was to make it as simple as possible.

But of course we can decide otherwise.

**Comment by neandr** [comment]
Holding all XPIs in that dir has the beauty a download/zip of the
project has everything. But also it will be a fat package and
unnecessary fat with the whole xpi history. Think for that reason github
has the "Releases".
For me a link in the ../xpi directory would be more that enough, if at all


Am 19.03.20 um 13:12 schrieb John Bieling:

**Comment by Sylvain303** [comment]
Hi @jobisoft 

Thanks for releasing!

I downloaded from the github releases thanks, and installed from file.xpi

When the extension asked to restart TB:
 
> Thunderbird had a problem and crashed.
> 
> To help us diagnose and fix the problem, you can send us a crash report.

A second restart from the crash report, succeed.

To you want me to do some more testing?
Should I report more missing feature about this extension?
Ex: updating the Subject: when inside the mail Tab (Enter on a email) 

Regards,
Sylvain.

**Comment by jobisoft** [comment]
Thanks for the report! If you find the time, you could try if the crash is reproducible: uninstall, restart, install again -> crash every time?

Regarding features: currently we want to check if all features of the old version work in v3.x. Is the missing feature you mentioned just missing in v3.x but present in v2.x? In either case you can create a new issue and provide the details, but I think we will not add entirely new features now, but only after the update for TB78 has been finished. But this is just my own opinion.

**Comment by RealRaven2000** [comment]
> 
> 
> @neandr : The idea behind having them in the repository directly: If you clone it, you have the XPI as well. I added downloadlinks to the release page. So far, you where the only one, who added his XPI there directly :-)
> 
> I also added download instructions to the README, is that enough? What is missing?

I second gNeandr's opinion. xpi should be an asset in the releases list, it's not necessary to have it in the actual source tree. Remember you could commit something without building every time (but hopefully still have something that can actually build using the script). Releases however should always have an installable xpi in them.

**Comment by jobisoft** [comment]
Ok, changed it. 

I will miss the xpi in the repo however. Having to go to github.com to get it, req. an online connection, which is not always present when I am on the road. Manually building it req. staching local changes and switching tags. It makes a simple task rather complicated. Furthermore, I do not see a difference between a "download link" in the release page, which points to the xpi in the main tree, compared to an "asset link" which points to a special storage location on github.com. But well, I got outnumbered.

**Comment by jobisoft** [comment]
@Sylvain303 : You no longer need to install dev versions:
https://addons.thunderbird.net/addon/edit-email-subject/

We are now able to upload our released version to the origianl ATN page and you will get updates automatically again, if a new version has been released. But you need to remove your current dev version and install the official one from ATN.
